File: src/divesites/cove2.erl
-module(cove2).
-export([site_info/0]).
-include("include/divepredictor.hrl").

site_info() -> 
	#divesite{
		id="cove2", 
		name="Alki Seacrest Cove 2",
		noaaTideStationId="9447110",
		noaaCurrentStationId="PCT1681",
		location=#divelocation{address="1660 Harbor Ave SW Seattle, WA 98126"},
		find_solutions=fun find_solutions/2

	}.

find_solutions(Tides, Currents) ->
	io:fwrite("Solutions Finder for Cove 2 called.~n"),
	[#divesolution{siteId="cove2",time=date(),length=120}].

File: python/sites/cove2.py
from models import Current, DiveLocation, DiveSite, DiveSolution, Tide

site = DiveSite(
    id="cove2",
    name="Alki Seacrest Cove 2",
    tide_station_id="9447110",
    current_station_id="PUG1507",
    location=DiveLocation(address="1660 Harbor Ave SW Seattle, WA 98126"),
)


def find_solutions(_tides: list[Tide], currents: list[Current]) -> list[DiveSolution]:
    return [
        DiveSolution(
            site_id=site.id,
            time=c.dt,
            description=f"{c.type} ({c.magnitude}) - Cove 2 is always divable",
        )
        for c in currents
    ]
